Docs/source/Software/BSP.rst

```diff
@@ -158,4 +158,23 @@ Additional Considerations
     (to find the frequency of a given timer), & ``BSP_Timer_GetTicksElapsed()``. We could also change 
     the function names for each timer. Another thing to note is that one timer should not be used for
     multiple resources because if ``BSP_Timer_GetTicksElapsed()`` is called for one resource, it will
-    interfere with the time passed for the second resource.
\ No newline at end of file
+    interfere with the time passed for the second resource.
+
+Watchdog Timer BSP: Sijin Woo
+=================================
+
+Purpose 
+    The purpose of this timer is to make sure that the BPS has not stalled while running. If we are
+    in the Bare-Metal version of our code, that means it is stuck in a loop somewhere and is not able
```

Comment:
This might be unnecessary but I think you should replace "it" with "the BPS" for clarity.